Editor/Editors/CinemachineVirtualCameraBaseEditor.cs

```diff
@@ -147,6 +148,73 @@ protected void DrawHeaderInInspector()
             ExcludeProperty("Header");
         }
 
+#if CINEMACHINE_UNITY_INPUTSYSTEM
+        /// <summary>
+        /// Adds a button that adds CinemachineInputProvider component to the vcam with a default look control
+        /// (XY axis), if the vcam has at least one component or extension that requires input and the vcam does not
+        /// already have a CinemachineInputProvider component. For a component or extension to require input, the
+        /// component or extension needs to override RequiresInput in CinemachineComponentBase or CinemachineExtension
+        /// respectively.
+        /// <seealso cref="CinemachineInputProvider"/>
+        /// <seealso cref="CinemachineComponentBase"/>
+        /// <seealso cref="CinemachineExtension"/>
+        /// </summary>
+        protected void DrawInputProviderButton()
+        {
+            var vcamBase = (CinemachineVirtualCameraBase) target;
+            if (vcamBase.ParentCamera as CinemachineFreeLook != null)
+            {
+                return;
+            }
+            
+            var freelook = vcamBase as CinemachineFreeLook;
+            if (freelook != null)
+            {
+                CinemachineDefaultMouseInput.InputProviderButton(vcamBase.gameObject);
+            }
+            else // check if any component or extension requires input
+            {
+                CinemachineComponentBase[] components = null;
+                List<CinemachineExtension> extensions = null;
+                
+                var vcam = vcamBase as CinemachineVirtualCamera;
+                if (vcam != null)
+                {
+                    components = vcam.GetComponentPipeline();
+                    extensions = vcam.mExtensions;
+                }
+                
+#if CINEMACHINE_EXPERIMENTAL_VCAM
+                var newVcam = vcamBase as CinemachineNewVirtualCamera;
+                if (newVcam != null)
+                {
+                    var tempComponents = newVcam.ComponentCache.ToList();
+                    for (int i = tempComponents.Count - 1; i >= 0; --i)
+                    {
+                        if (tempComponents[i] == null)
+                        {
+                            tempComponents.RemoveAt(i);
+                        }
+                    }
+                    components = tempComponents.ToArray();
```

Comment:
I could also check if the component is null before checking RequiresInput in InputRequiredByComponentsOrExtensions. I chose this to avoid checking for null every frame, because experimental is probably off for more users.

But, I can also check for null.

Which is better in your opinion?

Reply:
Thanks for the feedback. I understand and I am going to rethink this!

Reply:
Eek! Indeed! I don't know why I did it the way it was. I guess it was too late. ¯\_༼ᴼ ل͜ᴼ༽_/¯ SORRY!!!